src/modules/traceModelContext.js

```diff
@@ -0,0 +1,99 @@
+import { SessionManager } from './sessionManager.js';
+
+class TraceModelContext {
+
+    #traceModel;
+
+    /**
+     * Get the default trace model or a project trace model.
+     *
+     * @param {SessionManager} sessionManager The session manager, used for connections to the Hansken servers
+     * @param {UUID} projectId The projectId to get the traceModel from, or undefined when accessing the default trace model.
+     */
+    constructor(sessionManager, projectId) {
+        this.sessionManager = sessionManager;
+        this.url = projectId ? `/projects/${projectId}/tracemodel`: `/tracemodel`;
+    }
+
+    /**
+     * Get the trace model.
+     *
+     * @returns The trace model
+     */
+    get = () => {
+        if (!this.#traceModel) {
+            return this.sessionManager.gatekeeper(this.url).then(SessionManager.json).then((traceModel) => {
+                this.#traceModel = traceModel;
+                return traceModel;
+            });
+        }
+        return Promise.resolve(this.#traceModel);
```

Comment:
The request could be performed multiple times. Probably better to assign the request promise to `#traceModel`
```suggestion
        if (!this.#traceModel) {
            this.#traceModel = this.sessionManager.gatekeeper(this.url).then(SessionManager.json);
        }
        return this.#traceModel;
```

Reply:
Good idea